$NetBSD: patch-af,v 1.10 2025/10/20 18:32:10 vins Exp $

* Add check for stropts.h header and updwtmpx func. 
* Add check for ut_session in struct utmpx.
* Fix broken Xlocale support
* Use posix_openpt() to avoid GNUisms. 

--- configure.orig	2003-03-26 06:07:26.000000000 +0000
+++ configure
@@ -9481,6 +9481,7 @@ fi
 
 
 
+
 for ac_header in \
 	assert.h \
 	fcntl.h \
@@ -9496,6 +9497,7 @@ for ac_header in \
 	sys/ioctl.h \
 	sys/select.h \
 	sys/sockio.h \
+	sys/stropts.h \
 	sys/strredir.h \
 	sys/time.h \
 	utmp.h \
@@ -12964,6 +12966,7 @@ for ac_func in \
 	on_exit \
 	nanosleep \
 	updwtmp \
+	updwtmpx \
 	ttyslot \
 
 do
@@ -13580,6 +13583,67 @@ cat >>confdefs.h <<\_ACEOF
 _ACEOF
 
 fi
+echo "$as_me:$LINENO: checking for session in utmpx struct" >&5
+echo $ECHO_N "checking for session in utmpx struct... $ECHO_C" >&6
+if test "${rxvt_cv_struct_utmpx_session+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+#include <sys/types.h>
+#include <utmpx.h>
+int
+main ()
+{
+struct utmpx utx; utx.ut_session;
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
+  (eval $ac_compile) 2>conftest.er1
+  ac_status=$?
+  grep -v '^ *+' conftest.er1 >conftest.err
+  rm -f conftest.er1
+  cat conftest.err >&5
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } &&
+	 { ac_try='test -z "$ac_c_werror_flag"
+			 || test ! -s conftest.err'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; } &&
+	 { ac_try='test -s conftest.$ac_objext'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; }; then
+  rxvt_cv_struct_utmpx_session=yes
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+rxvt_cv_struct_utmpx_session=no
+fi
+rm -f conftest.err conftest.$ac_objext conftest.$ac_ext
+fi
+echo "$as_me:$LINENO: result: $rxvt_cv_struct_utmpx_session" >&5
+echo "${ECHO_T}$rxvt_cv_struct_utmpx_session" >&6
+if test x$rxvt_cv_struct_utmpx_session = xyes; then
+
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_UTMPX_SESSION 1
+_ACEOF
+
+fi
 
 fi
 
@@ -14233,8 +14297,9 @@ _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
+#include <stdlib.h>
 #include <X11/Xlib.h>
-main() {
+int main() {
 char *p;
 if ((p = XSetLocaleModifiers("@im=none")) != NULL && *p)
 exit (XSupportsLocale() ? 0 : 1);
@@ -14433,8 +14498,8 @@ _ACEOF
 
 fi
 
-echo "$as_me:$LINENO: checking for getpt" >&5
-echo $ECHO_N "checking for getpt... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for posix_openpt" >&5
+echo $ECHO_N "checking for posix_openpt... $ECHO_C" >&6
 if test "${rxvt_cv_func_getpt+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -14452,7 +14517,7 @@ cat >>conftest.$ac_ext <<_ACEOF
 int
 main ()
 {
-(void)getpt();
+(void)posix_openpt(0);
   ;
   return 0;
 }
